$NetBSD: patch-ab,v 1.2 2010/05/29 03:18:54 obache Exp $

* Fixed NumPy as numeric package for user-destdir installation.
* Honor CFLAGS and LDFLAGS.

--- Src/MPI/compile.py.orig	2008-08-18 12:54:41.000000000 +0000
+++ Src/MPI/compile.py
@@ -5,7 +5,6 @@
 import distutils
 import distutils.sysconfig
 import os, sys
-from Scientific import N
 
 cfgDict = distutils.sysconfig.get_config_vars()
 
@@ -13,8 +12,16 @@ cfgDict = distutils.sysconfig.get_config
 mpicompiler = 'mpicc'
 sources='mpipython.c Scientific_mpi.c'
 
+ldflags=""
+if os.environ.has_key('LDFLAGS'):
+    ldflags = os.environ['LDFLAGS']
+
+cflags=""
+if os.environ.has_key('CFLAGS'):
+    cflags = os.environ['CFLAGS']
+
 extra_compile_args = ""
-if N.package == "NumPy":
+if True:
     arrayobject_h_include = os.path.join(sys.prefix,
                             "lib/python%s.%s/site-packages/numpy/core/include"
                                           % sys.version_info [:2])
@@ -32,9 +39,11 @@ if sys.platform == 'darwin':
             items[i] = os.path.join(frameworkdir[0], items[i])
     linkforshared = ' '.join(items)
 
-cmd = '%s %s -o mpipython -I%s %s %s -L%s -lpython%s %s %s' % \
+cmd = '%s %s %s -o mpipython -I../../Include %s -I%s %s %s -L%s -lmpich -lpython%s %s %s' % \
     (mpicompiler, 
+     ldflags,
      linkforshared,
+     cflags,
      cfgDict['INCLUDEPY'],
      extra_compile_args,
      sources,
